pkg/agent/endpoints/workload/handler.go

```diff
@@ -264,24 +267,28 @@ func (h *Handler) FetchX509Bundles(_ *workload.X509BundlesRequest, stream worklo
 	}
 }
 
-func sendX509BundlesResponse(update *cache.WorkloadUpdate, stream workload.SpiffeWorkloadAPI_FetchX509BundlesServer, log logrus.FieldLogger, allowUnauthenticatedVerifiers bool) error {
+func sendX509BundlesResponse(update *cache.WorkloadUpdate, stream workload.SpiffeWorkloadAPI_FetchX509BundlesServer, log logrus.FieldLogger, allowUnauthenticatedVerifiers bool, previousResponse *workload.X509BundlesResponse) (*workload.X509BundlesResponse, error) {
 	if !allowUnauthenticatedVerifiers && !update.HasIdentity() {
 		log.WithField(telemetry.Registered, false).Error("No identity issued")
-		return status.Error(codes.PermissionDenied, "no identity issued")
+		return nil, status.Error(codes.PermissionDenied, "no identity issued")
 	}
 
 	resp, err := composeX509BundlesResponse(update)
 	if err != nil {
 		log.WithError(err).Error("Could not serialize X509 bundle response")
-		return status.Errorf(codes.Unavailable, "could not serialize response: %v", err)
+		return nil, status.Errorf(codes.Unavailable, "could not serialize response: %v", err)
+	}
+
+	if reflect.DeepEqual(resp.GetBundles(), previousResponse.GetBundles()) {
```

Comment:
DeepEqual won't do the right thing here. We should use proto.Equal instead. We should also compare the full response instead of just the bundle fields (more robust in the face of additional fields).
```suggestion
	if proto.Equal(resp, previousResponse) {
```